BuildPropertiesPlugin/plugin/src/test/groovy/com/novoda/buildproperties/EntrySubject.groovy

```diff
@@ -0,0 +1,49 @@
+package com.novoda.buildproperties
+
+import com.google.common.truth.FailureStrategy
+import com.google.common.truth.Subject
+import com.google.common.truth.SubjectFactory
+import com.google.common.truth.Truth
+
+import javax.annotation.Nullable
+
+final class EntrySubject extends Subject<EntrySubject, Entry> {
+
+    private static final SubjectFactory<EntrySubject, Entry> FACTORY = new SubjectFactory<EntrySubject, Entry>() {
+        @Override
+        EntrySubject getSubject(FailureStrategy fs, Entry that) {
+            new EntrySubject(fs, that)
+        }
+    }
+
+    public static EntrySubject assertThat(Entry entry) {
+        Truth.assertAbout(FACTORY).that(entry)
+    }
+
+    private EntrySubject(FailureStrategy failureStrategy, @Nullable Entry subject) {
+        super(failureStrategy, subject)
+    }
+
+    public void willThrow(Class<? extends Throwable> throwableClass) {
+        try {
+            subject.value
```

Comment:
Yes. This line is equivalent to `getSubject().getValue()` that in turns [evaluates the closure](https://github.com/novoda/spikes/blob/master/BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/Entry.groovy#L33-L35) inside the `Entry`. At the moment the only implementation available [will throw an `IllegalArgumentException`](https://github.com/novoda/spikes/blob/master/BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/FilePropertiesEntries.groovy#L48) when the specified property is not found.